### PR TITLE
Update dokka.yml

### LINKED ISSUE #1823: [Documentation] Adding a docusaurus deployment GitHub action
**Is your feature request related to a problem? Please describe.**
- We would like to have clear document implementors' documentation for the FHIR Core product.  

**Describe the solution you'd like**
- [ ] Add the docusaurus GitHub action to deploy the FHIR Core documentation. 
    - The documentation to do this can be found [here](https://docusaurus.io/docs/deployment#deploying-to-github-pages)

**Acceptance criteria**
- [ ]  The FHIR core docusaurus should be set up correctly to allow the deployment of new documentation and design of the documentation site.

### PR DESCRIPTION
**Pull request by SebaMutuku**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1823

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: .github/workflows/dokka.yml [hunk 1]
```diff
@@ -34,6 +34,6 @@ jobs:
       - name: Deploy 🚀
         uses: JamesIves/github-pages-deploy-action@4.1.4
         with:
```
Comment (pld) [inline .github/workflows/dokka.yml:36]: we don't want to overwrite main, just comment out as mentioned here, https://github.com/opensrp/fhircore/issues/1823#issuecomment-1315555032

```suggestion
#      - name: Deploy 🚀
#        uses: JamesIves/github-pages-deploy-action@4.1.4
#        with:
```

File: .github/workflows/dokka.yml [hunk 2]
```diff
@@ -34,6 +34,6 @@ jobs:
...
-          branch: gh-pages # The branch the action should deploy to.
+          branch: main # The branch the action should deploy to.
           folder: android/docs # The folder the action should deploy.
```
Comment (pld) [inline .github/workflows/dokka.yml:38]:
```suggestion
#          branch: gh-pages # The branch the action should deploy to.
#          folder: android/docs # The folder the action should deploy.
```

**Review by pld — CHANGES_REQUESTED** [review]

**Comment by SebaMutuku** [comment]
Closing this as other alternatives were provided.

**Comment by SebaMutuku** [comment]
Closing this as other alternatives were provided.

**Pull request closed without merging**